src/uct/api/uct.h

```diff
@@ -1213,6 +1214,66 @@ struct uct_md_attr {
 };
 
 
+/**
+ * @ingroup UCT_MD
+ * @brief UCT MD memory attributes field mask
+ *
+ * The enumeration allows specifying which fields in @ref uct_md_mem_attr_t
+ * are present.
+ */
+enum uct_md_mem_attr_field {
+    UCT_MD_MEM_ATTR_FIELD_MEM_TYPE = UCS_BIT(0), /**< Indicate if memory type
+                                                      is populated. Types
+                                                      belong to one of
+                                                      @ref ucs_memory_type_t */
+    UCT_MD_MEM_ATTR_FIELD_SYS_DEV  = UCS_BIT(1)  /**< Indicate if details of
+                                                      system device backing
+                                                      the pointer is populated.
+                                                      @ref ucs_sys_device_t*/
+};
+
+
+/**
+ * @ingroup UCT_MD
+ * @brief  Memory domain attributes.
+ *
+ * This structure defines the attributes of a memory pointer which may
+ * include memory type of the pointer, the system device that backs the
+ * pointer depending on the bit fields populated in field_mask
+ */
+typedef struct uct_md_mem_attr {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_md_mem_attr_t. Note that the field maske is
+     * populated upon return from uct_md_mem_query and not set by user.
+     * Subsequent use of members of the structure are valid after ensuring that
+     * relevant bits in the field_mask rare set
+     */
+    uint64_t          field_mask;
+    ucs_memory_type_t mem_type;   /**< Is the type CPU memory or GPU memory, etc*/
+    ucs_sys_device_t  sys_dev;    /**< location of device pointer. eg: NUMA/GPU */
+} uct_md_mem_attr_t;
+
+
+/**
+ * @ingroup UCT_MD
+ * @brief Query attributes of a given pointer
+ *
+ * Return attributes such as memory type, and system device for the
+ * given pointer of specific length. @ref uct_md_mem_attr_t
+ *
+ * @param [in]     md          Memory domain to allocate memory on.
+ * @param [in]     address     The address of the pointer. Must be non-NULL
+ * @param [in]     length      Indicates the range of memory starting from (address) to
+ *                             (address + length) bytes. Must be >0.
```

Comment:
Length of memory region to examine. Must be nonzero.

Reply:
done